tasks/ceph_deploy.py

```diff
@@ -174,12 +195,12 @@ def build_ceph_cluster(ctx, config):
                 ceph_branch = '--{var}={val}'.format(var=var, val=val)
         node_dev_list = []
         all_nodes = get_all_nodes(ctx, config)
-        mds_nodes = get_nodes_using_roles(ctx, config, 'mds')
+        mds_nodes = get_nodes_using_role(ctx, 'mds')
         mds_nodes = " ".join(mds_nodes)
-        mon_node = get_nodes_using_roles(ctx, config, 'mon')
+        mon_node = get_nodes_using_role(ctx, 'mon')
         mon_nodes = " ".join(mon_node)
         new_mon = './ceph-deploy new'+" "+mon_nodes
-        install_nodes = './ceph-deploy install '+ceph_branch+" "+all_nodes
+        install_nodes = './ceph-deploy install ' + (ceph_branch if ceph_branch else "--dev=master") + " " + all_nodes
```

Comment:
The reason this change is here is because the existing task would give a nasty exception if there wasn't an explicit branch set in the config -- it's a quick bug fix rather than an outright behavioural change (you only hit this path if you would previously have seen an exception).

The idea of defaulting to master is to match the behaviour of the default ceph install task, i.e. that when you run ceph-qa-suite master without any overrides, it should run against ceph master.